typescript/packages/plugins/balmy/src/balmy.service.ts

```diff
@@ -0,0 +1,186 @@
+            };
+        };
+    };
+    provider: {
+        source: {
+            type: "http";
+            url: string;
+            supportedChains: number[];
+        };
+        defaultConfig: {
+            chainId: number;
+            rpcUrl: string;
+        };
+    };
+};
+
+type TokenBalance = {
+    [address: string]: string;
+};
+
+type TokenBalances = {
+    [token: string]: TokenBalance;
+};
+
+type ChainBalances = {
+    [chainId: string]: TokenBalances;
+};
+
+export class BalmyService {
+    private sdk;
+
+    constructor(config: BalmyConfig) {
+        this.sdk = buildSDK(config);
+    }
+
+    @Tool({
+        description: "Get token balances for an account",
+    })
+    async getBalances(walletClient: EVMWalletClient, parameters: GetBalanceParameters) {
+        const chainid = await walletClient.getChain().id;
+
+        const balances = await this.sdk.balanceService.getBalances({
+            tokens: [
+                {
+                    chainId: chainid,
+                    account: parameters.account,
+                    token: parameters.token,
+                },
+            ],
+        });
+
+        return Object.entries(balances).reduce((acc: ChainBalances, [chainId, tokens]) => {
+            acc[chainId] = Object.entries(tokens).reduce((tokenAcc: TokenBalances, [token, balance]) => {
+                tokenAcc[token] = Object.entries(balance).reduce((balAcc: TokenBalance, [address, amount]) => {
+                    balAcc[address] = amount.toString();
+                    return balAcc;
+                }, {});
+                return tokenAcc;
+            }, {});
+            return acc;
+        }, {});
+    }
+
+    @Tool({
+        description: "Get token allowance for my account",
+    })
+    async getAllowances(walletClient: EVMWalletClient, parameters: GetAllowanceParameters) {
+        return await this.sdk.allowanceService.getAllowanceInChain({
+            chainId: parameters.chainId,
+            token: parameters.token,
+            owner: parameters.owner,
+            spender: parameters.spender,
+        });
+    }
+
+    @Tool({
+        description: "Get quotes for a token swap",
+    })
+    async getQuote(walletClient: EVMWalletClient, parameters: GetQuoteParameters) {
+        const chainid = await walletClient.getChain().id;
+
+        const request: QuoteRequest = {
+            chainId: chainid,
+            sellToken: parameters.tokenIn,
+            buyToken: parameters.tokenOut,
+            order:
+                parameters.order.type === "sell"
+                    ? {
+                          type: "sell",
+                          sellAmount: parameters.order.Amount,
+                      }
+                    : {
+                          type: "buy",
+                          buyAmount: parameters.order.Amount,
+                      },
+            slippagePercentage: parameters.slippagePercentage,
+            gasSpeed: parameters.gasSpeed,
```

Comment:
This is not specified in the parameters. I'd remove

Reply:
was optional , removed